Bottom: 477bc0ef1365d501b29b532a75ce3704e86043e6
Top:    060c25bd697a3483bc1bf3e2da81c8d2cd6a8ec0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-22 20:01:48 +0000

Refresh of hid-gtk-gl-add-hacky-code-to

---

diff --git a/src/polygon.c b/src/polygon.c
index d6996b7..38245ec 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2134,11 +2134,11 @@ POLYAREA *board_outline_poly (void)
 
     } while ((check = check->f) != clipped);
 
-    /* If the piece is inside an odd number of others, delete it */
-    if ((count & 1) == 0)
-      pieces_to_delete = g_list_prepend (pieces_to_delete, piece);
-    else
+    /* If the piece is inside an odd number of others, keep it */
+    if ((count & 1) == 1)
       any_pieces_kept = true;
+    else
+      pieces_to_delete = g_list_prepend (pieces_to_delete, piece);
 
   } while ((piece = piece->f) != clipped);
